prime-router/src/main/kotlin/serializers/Hl7Serializer.kt

```diff
@@ -193,7 +195,17 @@ class Hl7Serializer(
             }
 
             if (it.isNotBlank()) {
-                nextMessage.append("$it\r")
+                if (it.length + nextMessage.length <= nextMessage.capacity()) {
+                    if (it.length > MESSAGE_SEGMENT_LENGTH) {
+                        logger.warn("Message segment ${it.substring(0,3)} longer than 4000 characters, truncating")
+                        nextMessage.append("${it.substring(0, MESSAGE_SEGMENT_LENGTH)}\r")
+                    } else {
+                        nextMessage.append("$it\r")
+                    }
+                } else {
+                    logger.error("Message too long, returning.")
+                    error("Message too long")
+                }
             }
```

Comment:
Do we need to add Unit tests to test these changes?

Reply:
I started to but couldn't figure out how to actually call the method directly. No matter what I did I just couldn't call it in the test. Do you have an idea of how to do this?